costmap_2d/include/costmap_2d/costmap_2d.h

```diff
@@ -410,14 +410,14 @@ class Costmap2D
     inline void raytraceLine(ActionType at, unsigned int x0, unsigned int y0, unsigned int x1, unsigned int y1,
                              unsigned int max_length = UINT_MAX)
     {
-      int dx = x1 - x0;
-      int dy = y1 - y0;
+      int dx = static_cast<int>(x1 - x0);
```

Comment:
it would be better to detect whether this is a negative value because this would yield incorrect results if that were the case.

Reply:
@mbittarelli6river may be using ---> static_cast<int>(x1) - static_cast<int>(x0) ??

Reply:
the problem still exists. you can subtract 2 unsigned ints and come out with a negative number. when you cast an int to unsigned int you can still run into trouble

Reply:
so if we come up with a negative number here.. it will be assigned to an "int dx" here and here we are casting from unsigned int to int.. not otherwise..  so "int dx" can store a negative value. does it make sense? or I am not getting the issue at all..

Reply:
this is my bad. the code change here is fine